src/analysis/processing/qgsalgorithmexporttospatialite.cpp

```diff
@@ -0,0 +1,229 @@
+
+/***************************************************************************
+                         qgsalgorithmexporttospatialite.cpp
+                         ---------------------
+    begin                : April 2025
+    copyright            : (C) 2025 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmexporttospatialite.h"
+#include "qgsabstractdatabaseproviderconnection.h"
+#include "qgsproviderregistry.h"
+#include "qgsprovidermetadata.h"
+#include "qgsvectorlayerexporter.h"
+
+///@cond PRIVATE
+
+QString QgsExportToSpatialiteAlgorithm::name() const
+{
+  return QStringLiteral( "importintospatialite" );
+}
+
+QString QgsExportToSpatialiteAlgorithm::displayName() const
+{
+  return QObject::tr( "Export to SpatiaLite" );
+}
+
+QStringList QgsExportToSpatialiteAlgorithm::tags() const
+{
+  return QObject::tr( "export,import,spatialite,table,layer,into,copy" ).split( ',' );
+}
+
+QString QgsExportToSpatialiteAlgorithm::group() const
+{
+  return QObject::tr( "Database" );
+}
+
+QString QgsExportToSpatialiteAlgorithm::groupId() const
+{
+  return QStringLiteral( "database" );
+}
+
+QString QgsExportToSpatialiteAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Exports a vector layer to a SpatiaLite database, creating a new table." );
+}
+
+QString QgsExportToSpatialiteAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Exports a vector layer to a SpatiaLite database, creating a new table." );
+}
+
+Qgis::ProcessingAlgorithmFlags QgsExportToSpatialiteAlgorithm::flags() const
+{
+  return QgsProcessingAlgorithm::flags() | Qgis::ProcessingAlgorithmFlag::NoThreading;
```

Comment:
looks like the ONLY reason this is flagged as no threading is because of the vector layer "database" parameter, and the retrieval of "layer->dataProvider()->dataSourceUri();". We could easily do that in a prepare step to avoid the no threading requirement.